### PR TITLE
Add support for C++ std::any

### LINKED ISSUE #2633: std::any
I just wonder, how difficult it is to add std::any support to cython?!

### PR DESCRIPTION
**Pull request by EpigeneMax**
closes #2633

Not much to say in fact...
Added an `any.pxd` to `Cython/Includes/libcpp/` and a test inspired by other STL tests.

### CODE REVIEW

**Review by da-woods — COMMENTED** [review]
In principle Cython does know about typeid too I think so we could support `type`. But it's possible it's sufficiently special cased that it's hard to support

File: Cython/Includes/libcpp/any.pxd [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from libcpp cimport bool
+
+cdef extern from "<any>" namespace "std" nogil:
+    cdef cppclass any:
+        any()
+        any(any&)
+        void reset()
+        bool has_value()
+        T& emplace[T](...)
```
[the remark this answers is not in the document]
Comment (EpigeneMax) [reply Cython/Includes/libcpp/any.pxd]: The default constructor, `has_value`, `reset`, `type` and `swap` are all marked `noexcept`.
I added `except +` to all other methods.

**Comment by EpigeneMax** [comment]
I had not noticed that Cython supported `type_info`. I added the support for `type` (also in the tests).

**Comment by da-woods** [comment]
Thanks @EpigeneMax

**Pull request merged**